packages/admin/admin-icons/generate-icons.ts

```diff
@@ -60,15 +63,24 @@ const main = async () => {
 
 const getComponentName = (fileName: string) => pascalCase(fileName.split(".")[0], { transform: pascalCaseTransformMerge });
 
-const getPathData = (icon: Icon) => {
+const getSVGData = (icon: Icon) => {
     const fileContents = readFileSync(icon.path);
-    const parsedXml = new XMLParser({ ignoreAttributes: false, attributeNamePrefix: "" }).parse(fileContents.toString());
+    const parsedXml = new XMLParser({ ignoreAttributes: false }).parse(fileContents.toString());
 
-    if (parsedXml?.svg?.path?.d === undefined) {
-        throw new Error(`The file ${icon.name} must contain a <path> element with a d attribute`);
-    }
+    removeFillAttributes(parsedXml.svg);
+    return parsedXml.svg;
+};
 
-    return parsedXml.svg.path.d;
+const removeFillAttributes = (obj: any) => {
+    if (typeof obj === "object") {
+        Object.keys(obj).forEach((key) => {
+            if (key === "@_fill") {
+                delete obj[key];
+            } else if (typeof obj[key] === "object") {
+                removeFillAttributes(obj[key]);
+            }
+        });
+    }
```

Comment:
Could you make this immutable by creating and returning a new object instead of mutating the current object?

Reply:
I found a method to integrate the removal of the fill attribute into the XMLParser, therefor the whole function is not needed anymore. I deleted it

Reply:
Perfect 👌🏼